client/src/services/api.service.ts

```diff
@@ -1,15 +1,52 @@
 import axios from "axios";
 
-console.log("API SERVICE");
-
 const API_BASE = import.meta.env.DEV ? "http://localhost:10000/api" : "/api";
 
+export type AuthResponse = {
+  token: string;
+  user: {
+    id: string | number;
+    email: string;
+    fullName?: string;
+  };
+};
+
+const apiClient = axios.create({
+  baseURL: API_BASE,
+  withCredentials: true,
+});
+
+export const setAuthToken = (token: string | null) => {
```

Comment:
Since the header will be sent automatically through a cookie, this will be unnessary